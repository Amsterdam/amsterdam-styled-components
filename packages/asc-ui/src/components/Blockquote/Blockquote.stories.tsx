import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Blockquote from './Blockquote'

storiesOf('Atoms/Typography/Blockquote', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('default style', () => (
    <Blockquote>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Debitis deleniti
      dicta doloremque eveniet explicabo id ipsum omnis placeat rerum suscipit.
    </Blockquote>
  ))
