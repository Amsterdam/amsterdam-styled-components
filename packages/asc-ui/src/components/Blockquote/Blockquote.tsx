import React from 'react'
import BlockquoteStyle, { Props } from './BlockquoteStyle'

export { Props }

const Blockquote: React.FC<Props &
  React.HTMLAttributes<HTMLParagraphElement>> = ({
  children,
  ...otherProps
}) => <BlockquoteStyle {...otherProps}>{children}</BlockquoteStyle>

export default Blockquote
