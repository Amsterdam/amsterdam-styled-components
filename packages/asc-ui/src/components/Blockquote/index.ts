export { default as BlockquoteStyle } from './BlockquoteStyle'
export { default } from './Blockquote'
