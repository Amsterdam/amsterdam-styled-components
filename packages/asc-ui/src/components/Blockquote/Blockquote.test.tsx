import { render } from '@testing-library/react'
import Blockquote from './Blockquote'

describe('Blockquote', () => {
  it('should render', () => {
    const { container } = render(<Blockquote>Foo</Blockquote>)
    expect(container.firstChild).toBeDefined()
  })
})
