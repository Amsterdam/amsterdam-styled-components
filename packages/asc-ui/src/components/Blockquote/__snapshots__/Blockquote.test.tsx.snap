// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote should render 1`] = `
.c0 {
  margin: 0;
  color: #000000;
  font-weight: 400;
  font-size: 16px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 22px;
  margin-bottom: 20px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-bottom: 20px;
  padding: 4px 16px;
  border-left: 2px solid #B4B4B4;
  font-style: italic;
}

.c0 p {
  margin-bottom: 0;
}

@media screen and (min-width:540px) {
  .c0 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    font-size: 18px;
    line-height: 25px;
  }
}

<p
  class="c0"
>
  Foo
</p>
`;
